cloudinit/config/cc_scripts_per_once.py

```diff
@@ -5,30 +5,34 @@
 # Author: Juerg Haefliger <juerg.haefliger@hp.com>
 #
 # This file is part of cloud-init. See LICENSE file for license information.
+"""Scripts Per Once: Run one time scripts"""
 
-"""
-Scripts Per Once
-----------------
-**Summary:** run one time scripts
+import os
+
+from cloudinit import subp
+from cloudinit.config.schema import MetaSchema, get_meta_doc
+from cloudinit.distros import ALL_DISTROS
+from cloudinit.settings import PER_ONCE
 
+frequency = PER_ONCE
+MODULE_DESCRIPTION = """\
 Any scripts in the ``scripts/per-once`` directory on the datasource will be run
 only once. Changes to the instance will not force a re-run. The only way to
 re-run these scripts is to run the clean subcommand and reboot. Scripts will
 be run in alphabetical order. This module does not accept any config keys.
-
-**Internal name:** ``cc_scripts_per_once``
-
-**Module frequency:** per once
-
-**Supported distros:** all
 """
 
-import os
-
-from cloudinit import subp
-from cloudinit.settings import PER_ONCE
-
-frequency = PER_ONCE
+meta: MetaSchema = {
+    "id": "cc_scripts_per_once",
+    "name": "Scripts Per Once",
+    "title": "Run one time scripts",
+    "description": MODULE_DESCRIPTION,
+    "distros": [ALL_DISTROS],
```

Comment:
I don't think at the moment any of our modules [can avoid defining a module level attribute distros](https://warthogs.atlassian.net/browse/SC-888) because the [cloudinit.config._fixup_modules sets mod.distros = [] when absent](https://github.com/canonical/cloud-init/blob/main/cloudinit/config/__init__.py#L40-L41) and if mod.frequency is not definied cloud-init it'll be set to PER_INSTANCE (so good that you left this PER_ONCE as module-level attr

Reply:
Actually strike that on distros comment for modules with a distros = ["all"] case. when a config module doesn't define mod.distros -> fixup_modules sets mod.distros = [] -> [cloudinit.stages only checks/filters distros is distros is non-empty :)](https://github.com/canonical/cloud-init/blob/main/cloudinit/stages.py#L1119-L1129). Ok so modules which claim support of "all" distros can avoid defining mod.distros attribute as it'll be an inactive filter in that case.